### PR TITLE
Update CWSPR telemetry client as per API model changes.

### PR DESCRIPTION
**Pull request by saggarn**
CWSPR send telemetry API now supports 3 new event types: ChatAddMessageEvent, ChatInteractWithMessageEvent and ChatUserModificationEvent. This PR updates the client to capture these events using these specific event types instead of generic MetricData type.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ppiliaev — APPROVED** [review]

**Comment by leigaol** [comment]
Please fix the required Github CI checks including unit test. Thank you!

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt [hunk 1]
```diff
@@ -94,36 +99,30 @@ class TelemetryHelper(private val context: AmazonQAppInitContext, private val se
             cwsprChatSourceLinkCount = response.relatedSuggestions?.size,
             cwsprChatReferencesCount = 0, // TODO
             cwsprChatFollowUpCount = response.followUps?.size,
-            cwsprChatTimeToFirstChunk = getResponseStreamTimeToFirstChunk(response.tabId),
-            cwsprChatTimeBetweenChunks = "", // getResponseStreamTimeBetweenChunks(response.tabId), //TODO: allow '[', ']' and ',' chars
+            cwsprChatTimeToFirstChunk = getResponseStreamTimeToFirstChunk(response.tabId).toInt(),
+            cwsprChatTimeBetweenChunks = "[${getResponseStreamTimeBetweenChunks(response.tabId).joinToString(",")}]",
```
Comment (leigaol) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt:103]: Can you provide evidence for:

1. This change has been tested end to end with service. The API call responded with 200. 
2. The field `cwsprChatTimeBetweenChunks` can be de-serialized and it has the same schema as the one defined in https://github.com/aws/aws-toolkit-vscode/pull/4200/files

Comment (saggarn) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt:103]: Here are some request Ids generated from RTS prod:
1. Add message event: `effada91-43f0-4dcb-8098-7ca3e914bf19`
2. Interact with message event: `6b24391c-f526-4db2-8313-fe209e8cb14b`
3. User modification event: `31030da3-b944-4c25-945c-c734064760ed`

**Review by leigaol — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt (hunk 1 above)
Comment (saggarn) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt:102]: Shared some request Ids offline.

**Review by manodnyab — APPROVED** [review]

**Pull request merged**